testbench/common.py

```diff
@@ -737,10 +738,55 @@ def limited_response_fn(data):
     return limited_response_fn
 
 
+def grpc_handle_retry_test_instruction(database, request, context, method):
+    test_id = None
+    if context is not None:
+        if hasattr(context, "invocation_metadata") and isinstance(
```

Comment:
If context isn't these things, should there be a ValueError?

Reply:
If the context isn't these things, the retry handler is expected to not do anything, and simply let the request pass through by returning the default response.